source/solvers/flow_control.cc

```diff
@@ -111,14 +111,36 @@ FlowControl<dim>::calculate_beta(const std::pair<double, double> &flow_rate,
   flow_rate_1n = flow_rate_n;
 }
 
-
 template <int dim>
-Tensor<1, dim>
-FlowControl<dim>::get_beta()
+void
+FlowControl<dim>::save(std::string prefix)
 {
-  return beta;
+  std::string   filename = prefix + ".flowcontrol";
+  std::ofstream output(filename.c_str());
+  output << "Flow control" << std::endl;
+  output << "Previous_beta " << beta_n << std::endl;
+  output << "Previous_flow_rate " << flow_rate_1n << std::endl;
+  output << "No_force " << no_force << std::endl;
+  output << "Threshold_factor " << threshold_factor << std::endl;
 }
 
+template <int dim>
+void
+FlowControl<dim>::read(std::string prefix)
+{
+  std::string   filename = prefix + ".flowcontrol";
+  std::ifstream input(filename.c_str());
+  if (!input)
+    {
+      throw("Unable to open file");
```

Comment:
You have to throw an exception here and not just throw a string.
For example `throw std::runtime_error("Text")`
Or you can use AssertThrow
https://www.dealii.org/current/doxygen/deal.II/group__Exceptions.html#gafc0ca7ad85b3ebd64e8e51689ac85caf

Reply:
Ok, I just copy/paste it from Simulation control, it should be modified there too then.